### PR TITLE
Add clarification to metadata topic

### LINKED ISSUE #252: Clarify metadata topic
Add clarification to the [front matter](https://grafana.com/docs/writers-toolkit/writing-guide/front-matter/) topic about which metadata are required and add MenuTitle to the topic.

### PR DESCRIPTION
**Pull request by JStickler**
Fixes #252

### CODE REVIEW

File: docs/sources/writing-guide/front-matter/index.md
Comment (jdbaldry) [inline docs/sources/writing-guide/front-matter/index.md:47]: Aliases use URL paths, not file paths.

I also chose to remove the specific mention of HTML redirects because we should revise that information. The actual production website does not use HTML redirects but HTTP redirects. 
However, the local webserver still uses HTML redirects.
In practice, they result in the same behavior.
 
```suggestion
Used to create redirects from the previous URL to the new URL when a page changes or moves. 
As a best practice, when you rename or move files, you should create an alias with a reference to the previous URL path create a redirect from the old URL to the new URL. For more information, refer to [Hugo aliases](#hugo-aliases).
```

File: docs/sources/writing-guide/front-matter/index.md
Comment (jdbaldry) [inline docs/sources/writing-guide/front-matter/index.md:51]: Fixes typos.

```suggestion
Used to provide the short description of the topic to search engines, including the search feature in the Grafana documentation site.
The description is also displayed on social media, such as Twitter, to provide a clue to users about what the page includes.
```

File: docs/sources/writing-guide/front-matter/index.md
Comment (jdbaldry) [inline docs/sources/writing-guide/front-matter/index.md:66]: Fixes typo.

```suggestion
The website uses keywords to generate links to related pages in the _Related content_ sections.
```

File: docs/sources/writing-guide/front-matter/index.md
Comment (jdbaldry) [inline docs/sources/writing-guide/front-matter/index.md:126]:
```suggestion
Use the `weight` front matter to specify the placement of a topic within the left-hand sidebar on https://grafana.com.
Smaller numbers place the topic higher in the guide or section of the guide.
Pages with the same weight will be displayed in alphabetical order.
```

No weight is not an exceptional case, it gets the default weight value and all pages with the same weight are sorted alphabetically.

**Review by jdbaldry — CHANGES_REQUESTED** [review]

File: docs/sources/writing-guide/front-matter/index.md
Comment (JStickler) [reply docs/sources/writing-guide/front-matter/index.md:127]: Hmmm, I need to figure out how to reword that.  What I was aiming for was that if you don't specify weights at all that pages are displayed in alphabetical order.  What is the default weight value?  I didn't realize there was one?

Comment (jdbaldry) [reply docs/sources/writing-guide/front-matter/index.md:127]: I believe the default is 0.

Comment (jdbaldry) [reply docs/sources/writing-guide/front-matter/index.md:127]: I double checked because I wasn't certain. It turns out I was completely wrong, sorry about that. According to the maintainer, pages are sorted by `weight` only if it set and then by `linkTitle` or `title`. So In that case, I think your update is better than my suggestion.

https://discourse.gohugo.io/t/how-do-i-sort-by-weight-then-title/18297/2

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [reply docs/sources/writing-guide/front-matter/index.md:47]: Is this a best practice or a requirement? What's the scenario where you've renamed or moved a page and you don't want an alias?

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [inline docs/sources/writing-guide/front-matter/index.md:49]: Not sure if the "required" should come after the item rather than before. I was gonna see what it looked like in preview, but you need to merge your branch with main to get the updates that allow the make docs command to work.

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [inline docs/sources/writing-guide/front-matter/index.md:51]: I immediately thought you were talking about the feature called "Search" in Grafana. Mixing language here confuses the issue I think. 
```suggestion
Used to provide the short description of the topic to search engines, including the search engine used in the Grafana documentation site.
```

File: docs/sources/writing-guide/front-matter/index.md
Comment (JStickler) [reply docs/sources/writing-guide/front-matter/index.md:49]: My aim was to have "required" visible in the right TOC.  I don't really care if it's before or after the heading, but I've usually seen the "required" or "optional" at the beginning so that people who scan don't miss it, and if things wrap it doesn't get lost.

I couldn't get `make-docs` to run last week, so I haven't seen how this looks in a preview yet.  I should check and see if Jack fixed the script.

Comment (imatwawana) [reply docs/sources/writing-guide/front-matter/index.md:49]: He has! I think you might have to merge your branch with main to get the fix.

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [reply docs/sources/writing-guide/front-matter/index.md:47]: I guess I just want stronger language. Maybe talk both scenarios. If it's possible for there to be a working redirect (you move an entire file or rename it) set up a redirect. If you've deleted content and there's no corresponding content or it's moved to within another page then don't. I just don't want people left with the idea that redirects are a nice-to-have. They're a requirement unless you can't set one up feasibly or you're gonna create bad 404s.

Comment (JStickler) [reply docs/sources/writing-guide/front-matter/index.md:47]: Ah, gotcha.  Let me think about rewording.  We definitely need **_something_**, because currently we don't say anything about why you would need an alias, and we definitely want to encourage people to create them.

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [reply docs/sources/writing-guide/front-matter/index.md:49]: Yeah looking at this in a local build, I find the list less scannable seeing the word in parentheses first and then the actual thing element i want information about second. recommend moving (required) to after the front matter element.
```suggestion
### description (required) 
```

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [inline docs/sources/writing-guide/front-matter/index.md:114]:
```suggestion
### title (required) 
```

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [inline docs/sources/writing-guide/front-matter/index.md:122]: Suggest moving this up to line 118 since not knowing this will cause you pain.

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [inline docs/sources/writing-guide/front-matter/index.md:126]:
```suggestion
By default, topics are displayed in alphabetical order by `title`.
```

File: docs/sources/writing-guide/front-matter/index.md
Comment (imatwawana) [inline docs/sources/writing-guide/front-matter/index.md:128]:
```suggestion
Use `weight` to specify a different topic order within the left-hand sidebar on https://grafana.com. Smaller numbers place the topic earlier in the guide or section of the guide. Pages with the same weight are displayed in alphabetical order.
```

**Review by imatwawana — APPROVED** [review]
I've left a few comments but otherwise, this looks good to me!

**Review by jdbaldry — APPROVED** [review]
LGTM once @imatwawana's comments are resolved.

**Review by josmperez — APPROVED** [review]
At quick look over, LGTM.

**Pull request merged**
